fix(APP-4559) and chore(APP-4614): implement node tooling version enforcement; adjust collapsible height

tyhonchik opened the pull request:
**chore(APP-4614)**: Enforces Nodejs version checks by adding yarn plugin and nvmrc

**fix(APP-4559)**: showOverlay breaks the collapsible height parameters and looks ugly - increased collapsedLines in Dao proposal page description

# Description

Task: [APP-4559](https://aragonassociation.atlassian.net/browse/APP-4559) showOverlay breaks the collapsible height parameters and looks ugly.
- Increased `collapsedLines` up to 8 as suggested in the task

Task: [APP-4614](https://aragonassociation.atlassian.net/browse/APP-4614) Enforce Nodejs version checks.
- Added .nvmrc, yarn plugin to check NodeJS version compatibility and fixed docs so new developers and new machines automatically use right NodeJS version.

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Developer Checklist:

- [x] Manually smoke tested the functionality locally
- [x] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [ ] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [x] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [x] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] Tested locally that all Acceptance Criteria or Expected Outcomes are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project


[APP-4559]: https://aragonassociation.atlassian.net/browse/APP-4559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-4614]: https://aragonassociation.atlassian.net/browse/APP-4614?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sepehr2github commented on packages/web-app/src/containers/configureWithdraw/index.tsx:
```diff
@@ -115,25 +115,27 @@ const ConfigureWithdrawForm: React.FC<ConfigureWithdrawFormProps> = ({
+          setValue(`actions.${actionIndex}.tokenPrice`, apiData.price);
         }
+
+        if (!apiData) {
```
Shouldn't be if !apiData && chainData?

sepehr2github reviewed (commented):
Thanks, looks good to me, I just left a simple suggestion, please let me know what you think

Quick thing, Have you disabled the cache because the deposit didn't come for me after submitting the transaction

Fabricevladimir commented:
@sepehr2github 
> Quick thing, Have you disabled the cache because the deposit didn't come for me after submitting the transaction

Cache is virtually untouched. Also, I can't reproduce. The deposit came in immediately

sepehr2github reviewed (approved):
thanks for the changes

tyhonchik committed:
chore(APP-4614): Enforces Nodejs version checks by adding yarn plugin and nvmrc

fix(APP-4559): showOverlay breaks the collapsible height parameters and looks ugly:
- increased collapsedLines in Dao proposal page description

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/aragon/app?pullRequest=783) <br/>All committers have signed the CLA.

thekidnamedkd commented on .yarn/plugins/engines-enforcer.cjs:
```diff
@@ -0,0 +1,78 @@
+'use strict';
```
This looks cool. Been having some node/yarn issues lately myself. Is it related to the task however? Maybe we can document how this is helpful

thekidnamedkd replied:
Disregard mostly, I see the ticket is two tasks from the description. We also document that in the title as fix(APP-4614, APP-4559).

tyhonchik replied:
A fair comment, thanks!
But when I saw that the fix was in one character, I decided to combine the commits into one PR.

I'll also change the PR name and improve the description to make it more clear which tickets are involved.

thekidnamedkd reviewed (commented):
Okay disregard my comment, I see in the description now this is two tickets and personally I think it makes sense since the other fix is so tiny, but then I would actually make the enforcer the focus of the ticket title and documentation. So like "feat: implement node tooling version enforcement, adjust collapsible height" or something. Looks great otherwise and super useful.

thekidnamedkd commented:
Totally. I would do the same.

I do wonder though re: the fix if maybe there’s a way on GovKit component
impl to get the desired look while still having the line count actually
make sense visually with overlay. i.e lineCount is actually 2 lines instead
of just pushing the height so it appears that away.

---
🌻
kevin davis
kevin-davis.com


On Wed, Oct 1, 2025 at 19:24 Andrew ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .yarn/plugins/engines-enforcer.cjs
> <https://github.com/aragon/app/pull/783#discussion_r2395320336>:
>
> > @@ -0,0 +1,78 @@
> +'use strict';
>
> A fair comment, thanks!
> But when I saw that the fix was in one character, I decided to combine the
> commits into one PR.
>
> I'll also change the PR name and improve the description to make it more
> clear which tickets are involved.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aragon/app/pull/783#discussion_r2395320336>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APVQ3TQCFN6XZCLLAEKW3VL3VQE3VAVCNFSM6AAAAACIAKRQQSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTEOJQGI2DOMBXGY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

tyhonchik commented:
> Totally. I would do the same. I do wonder though re: the fix if maybe there’s a way on GovKit component impl to get the desired look while still having the line count actually make sense visually with overlay. i.e lineCount is actually 2 lines instead of just pushing the height so it appears that away.

Let me dig deeper into this

If it is better to make a fix to GovKit, I'll do it and will remove commit for increasing collapsedLines

milosh86 commented on src/modules/governance/pages/daoProposalDetailsPage/daoProposalDetailsPageClient.tsx:
```diff
@@ -149,7 +149,7 @@ export const DaoProposalDetailsPageClient: React.FC<IDaoProposalDetailsPageClien
                         <Page.MainSection title={t('app.governance.daoProposalDetailsPage.main.description.header')}>
                             <CardCollapsible
-                                collapsedLines={2}
+                                collapsedLines={8}
```
I don't think this works in a general way. It works here only because given example has 8 lines, but if you have more, than the same issue would occur.

tyhonchik commented:
I split this PR into 2:

- https://github.com/aragon/gov-ui-kit/pull/594
- https://github.com/aragon/app/pull/787
